services/autoscaling/src/simcore_service_autoscaling/models.py

```diff
@@ -212,3 +212,13 @@ class BufferPoolManager:
 
     def __repr__(self) -> str:
         return f"BufferPoolManager({dict(self.buffer_pools)})"
+
+    def flatten_buffer_pool(self) -> BufferPool:
+        """returns a flattened buffer pool with all the EC2InstanceData"""
+        flat_pool = BufferPool()
+
+        for buffer_pool in self.buffer_pools.values():
+            for f in fields(BufferPool):
+                getattr(flat_pool, f.name).update(getattr(buffer_pool, f.name))
```

Comment:
this is updating the lhs from the rhs set? both are sets right?

Reply:
yep